engines/suppliers/app/controllers/suppliers/api/v1/suppliers_controller.rb

```diff
@@ -0,0 +1,64 @@
+module Suppliers
+  module Api
+    module V1
+      class SuppliersController < ApplicationController
+        before_action :load_supplier, only: [:show, :update, :destroy]
+
+        # GET /api/v1/suppliers
+        #
+        def index
+          suppliers = SuppliersFinder.new(supplier_params[:group_id]).find
```

Comment:
To avoid unnecessary multiplication of classes, though, you can group all access to the database into a Repository (instead of making each query into a class)
```ruby
module Suppliers
  class Repository
    def model
      Supplier
    end

    def find_by_group(group_id)
      model.where(group_id: group_id)
    end
  end
end